libclingcon/src/clingcon.cc

```diff
@@ -167,6 +197,21 @@ template<class T, T min=std::numeric_limits<T>::min(), T max=std::numeric_limits
     throw std::invalid_argument("invalid argument");
 }
 
+[[nodiscard]] double parse_num(char const *begin, char const *end = nullptr, double min=std::numeric_limits<double>::lowest(), double max=std::numeric_limits<double>::max()) {
```

Comment:
It should be possible to write the template so that it applies to both integers and floats.

Reply:
I wouldn't know how, as you can't have float or double values as template parameters in C++ yet.